client/utils.c

```diff
@@ -759,3 +759,71 @@ TDNFGetCmdOpt(
     return dwError;
 }
 
+GSList* TDNFMergeList(GSList* listA, GSList* listB)
+{
+    GSList* mergedList = NULL;
+ 
+    // Base cases
+    if (listA == NULL) 
+        return (listB);
+    else if (listB == NULL)
+        return (listA);
+
+    //Compare the URL Preference to sort the URL List in descending order. 
+    TDNF_METALINK_URL *urlA = listA->data;
+    TDNF_METALINK_URL *urlB = listB->data;
+    if (urlA->preference >= urlB->preference) 
+    {
+        mergedList = listA;
+        mergedList->next = TDNFMergeList(listA->next, listB);
+    }
+    else 
+    {
+        mergedList = listB;
+        mergedList->next = TDNFMergeList(listA, listB->next);
+    }
+    return (mergedList);
+}
+
+void TDNFFrontBackSplit(GSList* currHead,
+                   GSList** frontRef, GSList** backRef)
+{
+    GSList* fastPtr;
```

Comment:
we can initialise it while declaring itself
GSList* slowPtr = currHead;
GSList* fastPtr = currHead->next;